### PR TITLE
[SPARK-27123][SQL][FOLLOWUP] Use isRenaming check for limit too.

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

This is a followup for https://github.com/apache/spark/pull/24049 to reduce the scope of pattern based on the review comments.

## How was this patch tested?

Pass the existing test.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
@cloud-fan , @gatorsmile , @maropu . This is the followup PR for your comments. Could you review this please?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala [hunk 1]
```diff
@@ -699,20 +703,13 @@ object CollapseProject extends Rule[LogicalPlan] {
...
-        g.copy(child = l.copy(child = p2.copy(projectList = newProjectList)))
-      }
-    case p1 @ Project(_, l @ LocalLimit(_, p2: Project)) =>
-      if (haveCommonNonDeterministicOutput(p1.projectList, p2.projectList)) {
-        p1
-      } else {
-        val newProjectList = buildCleanedProjectList(p1.projectList, p2.projectList)
-        l.copy(child = p2.copy(projectList = newProjectList))
-      }
+    case Project(l1, g @ GlobalLimit(_, limit @ LocalLimit(_, p2 @ Project(l2, _))))
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala]: This is already covered by `RemoveNoopOperators`, IIUC.

Comment (dongjoon-hyun) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala]: If there is `Alias`, It wasn't. That is the point of my previous PR. 

Please see `IsRenaming` logic here. This considers that case, too. 
- https://github.com/apache/spark/pull/24082/files#diff-a636a87d8843eeccca90140be91d4fafL764

Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala]: ah i see, makes sense

**Comment by cloud-fan** [comment]
LGTM

**Comment by dongjoon-hyun** [comment]
Thank you so much for review and approval, @cloud-fan .

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Pull request closed without merging**